vsintegration/src/FSharp.Editor/LanguageService/WorkspaceExtensions.fs

```diff
@@ -216,7 +216,7 @@ type Project with
 
     /// Find F# references in the given project.
     member this.FindFSharpReferencesAsync(symbol, onFound, userOpName) =
-        async {
-            for doc in this.Documents do
-                do! doc.FindFSharpReferencesAsync(symbol, (fun textSpan range -> onFound doc textSpan range), userOpName)
-        }
+        this.Documents
+        |> Seq.map (fun doc -> doc.FindFSharpReferencesAsync(symbol, (fun textSpan range -> onFound doc textSpan range), userOpName))
+        |> Async.Parallel
```

Comment:
This also adds a noticeable boost in performance and it seems to work fine. But still there might have been a reason this was done sequentially before.

Reply:
Do you think it's worth putting it behind the flag, in case there are issues for end users?

Reply:
Yeah we could put this in performance options.

Reply:
Btw. **onFound** is not thread safe, since it is a appending to a collection builder.
Can be turned thread safe when replacing the collection used underneath ( even something naive like a Blockingcollection pre-allocated to e.g. 1024 items would do)

Reply:
FYI: And we do already have parallelism on the level of projects (each project scanned in parallel).

Reply:
@T-Gro good catch! I've been mostly testing it with _Find all references_ which has hopefully a thread-safe `onFound`, at least it seemed to always find the same amount. But this one for sure needs to be changed.